File: src/tools/semgrep/eval-repos/synthetic/python/entrypoint_patterns.py
"""Test file for Elttam entrypoint discovery patterns.

Contains Flask, Django, and FastAPI entrypoint patterns.
"""

# Flask entrypoints
from flask import Flask, request, jsonify

app = Flask(__name__)


# ENTRYPOINT_DISCOVERY: Flask route
@app.route('/')
def index():
    return jsonify({"message": "Home page"})


# ENTRYPOINT_DISCOVERY: Flask route with methods
@app.route('/api/users', methods=['GET', 'POST'])
def users():
    if request.method == 'GET':
        return jsonify({"users": []})
    return jsonify({"created": True})


# ENTRYPOINT_DISCOVERY: Flask route with parameter
@app.route('/api/users/<int:user_id>')
def get_user(user_id):
    return jsonify({"id": user_id})


# ENTRYPOINT_DISCOVERY: Flask blueprint route
from flask import Blueprint

api = Blueprint('api', __name__)


@api.route('/products')
def products():
    return jsonify({"products": []})


# Django entrypoints (simulated patterns)
# ENTRYPOINT_DISCOVERY: Django URL pattern
# urlpatterns = [
#     path('api/orders/', views.orders, name='orders'),
# ]

# ENTRYPOINT_DISCOVERY: Django view function
def django_view(request):
    return {"data": "response"}


# ENTRYPOINT_DISCOVERY: Django class-based view
class DjangoListView:
    def get(self, request):
        return {"items": []}

    def post(self, request):
        return {"created": True}


# Django REST Framework patterns
# ENTRYPOINT_DISCOVERY: DRF ViewSet
class UserViewSet:
    def list(self, request):
        return {"users": []}

    def retrieve(self, request, pk=None):
        return {"user": {"id": pk}}

    def create(self, request):
        return {"created": True}


# ENTRYPOINT_DISCOVERY: DRF APIView
class OrderAPIView:
    def get(self, request):
        return {"orders": []}

    def post(self, request):
        return {"order_id": 123}


# FastAPI entrypoints
from fastapi import FastAPI, Path, Query

fastapi_app = FastAPI()


# ENTRYPOINT_DISCOVERY: FastAPI route
@fastapi_app.get("/")
async def fastapi_root():
    return {"message": "Hello World"}


# ENTRYPOINT_DISCOVERY: FastAPI route with path parameter
@fastapi_app.get("/items/{item_id}")
async def fastapi_get_item(item_id: int = Path(...)):
    return {"item_id": item_id}


# ENTRYPOINT_DISCOVERY: FastAPI POST endpoint
@fastapi_app.post("/items/")
async def fastapi_create_item(item: dict):
    return item


# ENTRYPOINT_DISCOVERY: FastAPI with query parameters
@fastapi_app.get("/search/")
async def fastapi_search(q: str = Query(None)):
    return {"query": q}


# Tornado handler pattern
# ENTRYPOINT_DISCOVERY: Tornado RequestHandler
class TornadoHandler:
    def get(self):
        self.write({"data": "response"})

    def post(self):
        self.write({"created": True})


# AUDIT_INPUT_SINK: User input handling
@app.route('/api/process', methods=['POST'])
def process_input():
    user_data = request.get_json()  # AUDIT_INPUT_SINK
    return jsonify({"processed": user_data})


# AUDIT_FILE_UPLOAD: File upload endpoint
@app.route('/api/upload', methods=['POST'])
def upload_file():
    file = request.files['file']  # AUDIT_FILE_UPLOAD
    return jsonify({"filename": file.filename})

File: src/tools/semgrep/eval-repos/synthetic/javascript/entrypoint_patterns.js
/**
 * Test file for Elttam entrypoint discovery patterns.
 * Contains Express, Koa, and NestJS entrypoint patterns.
 */

const express = require('express');
const app = express();

// ENTRYPOINT_DISCOVERY: Express route
app.get('/', (req, res) => {
    res.json({ message: 'Home page' });
});

// ENTRYPOINT_DISCOVERY: Express route with parameter
app.get('/api/users/:id', (req, res) => {
    res.json({ id: req.params.id });
});

// ENTRYPOINT_DISCOVERY: Express POST route
app.post('/api/users', (req, res) => {
    res.json({ created: true, user: req.body });
});

// ENTRYPOINT_DISCOVERY: Express PUT route
app.put('/api/users/:id', (req, res) => {
    res.json({ updated: true, id: req.params.id });
});

// ENTRYPOINT_DISCOVERY: Express DELETE route
app.delete('/api/users/:id', (req, res) => {
    res.status(204).send();
});

// ENTRYPOINT_DISCOVERY: Express Router
const router = express.Router();

router.get('/products', (req, res) => {
    res.json({ products: [] });
});

router.post('/products', (req, res) => {
    res.json({ created: true });
});

app.use('/api', router);

// ENTRYPOINT_DISCOVERY: Express middleware
app.use('/api/protected', (req, res, next) => {
    // AUDIT_AUTH_CHECK: Authentication middleware
    if (!req.headers.authorization) {
        return res.status(401).json({ error: 'Unauthorized' });
    }
    next();
});

// Koa routes (simulated patterns)
// ENTRYPOINT_DISCOVERY: Koa route
const koaHandler = async (ctx) => {
    ctx.body = { message: 'Koa response' };
};

// Fastify routes (simulated patterns)
// ENTRYPOINT_DISCOVERY: Fastify route
const fastifyHandler = async (request, reply) => {
    return { message: 'Fastify response' };
};

// NestJS patterns (simulated)
// ENTRYPOINT_DISCOVERY: NestJS Controller
class UsersController {
    // ENTRYPOINT_DISCOVERY: NestJS route handler
    getAll() {
        return { users: [] };
    }

    // ENTRYPOINT_DISCOVERY: NestJS route with param
    getOne(id) {
        return { user: { id } };
    }

    // ENTRYPOINT_DISCOVERY: NestJS POST handler
    create(createUserDto) {
        return { created: true, user: createUserDto };
    }
}

// AUDIT_INPUT_SINK: User input handling
app.post('/api/process', (req, res) => {
    const userData = req.body; // AUDIT_INPUT_SINK
    res.json({ processed: userData });
});

// AUDIT_FILE_UPLOAD: File upload handling
const multer = require('multer');
const upload = multer({ dest: 'uploads/' });

app.post('/api/upload', upload.single('file'), (req, res) => {
    res.json({ filename: req.file.filename }); // AUDIT_FILE_UPLOAD
});

// AUDIT_SESSION: Session handling
app.get('/api/profile', (req, res) => {
    const session = req.session; // AUDIT_SESSION
    res.json({ user: session.user });
});

module.exports = { app, router, UsersController };
